.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,70 @@
+# This is a basic workflow to help you get started with Actions
+
+name: benchmark-check
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # This is too complicated to get working for now.
+      #
+      # # fetch the target branch from origin too (just the head)
+      # # and check it out into a local branch
+      # - name: Checkout PR and target branch
+      #   run: |
+      #     git fetch --no-tags --depth=1 origin ${{ github.base_ref }}
+      #     git fetch --no-tags --depth=1 origin ${{ github.head_ref }}
+
+      - name: Setup asv
+        run: |
+          pip install asv
+          cd benchmarks
+          asv machine --yes --machine gh-actions
```

Comment:
I'm nervous about `--machine gh-actions`. It would be good to make it clear that a set of results come from a specific GHA run, given the likelihood that GHA's `ubuntu-latest` spec will evolve over time, and the likelihood that we will subject results to increased scrutiny over time (e.g. through the ASV dashboard, or by downloading the `.asv/results` content.

So I'd prefer to use whatever machine name GHA provides. What do you think?

Reply:
I agree for storing long term results.  But this CI check only needs to be consistent with itself - everything is being thrown away at the end.

Reply:
But i'm happy to use the automatically ascribed name if that's simpler :+1:

Reply:
Yes I get that it's not currently a problem, I'm just worried about establishing an anti-pattern for later when we might forget or not be the ones expanding our ASV use.